include/seastar/core/iostream-impl.hh

```diff
@@ -152,26 +153,28 @@ future<> output_stream<CharType>::write(temporary_buffer<CharType> p) noexcept {
 template <typename CharType>
 future<temporary_buffer<CharType>>
 input_stream<CharType>::read_exactly_part(size_t n, tmp_buf out, size_t completed) noexcept {
-    if (available()) {
-        auto now = std::min(n - completed, available());
-        std::copy(_buf.get(), _buf.get() + now, out.get_write() + completed);
-        _buf.trim_front(now);
-        completed += now;
-    }
-    if (completed == n) {
-        return make_ready_future<tmp_buf>(std::move(out));
-    }
+    while (completed < n) {
```

Comment:
the reason why we passed `completed` to this function was that we wanted to call this function recursively. i think since we have the luxury of using coroutine, why not just drop the `completed` parameter?

Reply:
What's the opinion on such breaking API changes? I guess there's a tradeoff between making code cleaner and breaking api.

Reply:
it does not break the public API. it is but a private function.

Reply:
ultimately decided against removing this parameter, due to Avi's idea about fast path vs slow path.